src/com/amazon/ion/IonCursor.java

```diff
@@ -0,0 +1,83 @@
+package com.amazon.ion;
+
+import java.io.Closeable;
+
+/**
+ * A cursor over a stream of Ion data.
+ */
+public interface IonCursor extends Closeable {
+
+    /**
+     * Conveys the type of event that occurred as a result of operating on the cursor.
+     */
+    enum Event {
+
+        /**
+         * There is not enough data in the stream to complete the requested operation. The operation should be retried
+         * when more data is available.
+         */
+        NEEDS_DATA,
+
+        /**
+         * The cursor has completed an operation (e.g. `stepIntoContainer()`) and requires another instruction in order
+         * to position itself on the next value.
+         */
+        NEEDS_INSTRUCTION,
+
+        /**
+         * The cursor is positioned on a scalar value.
+         */
+        START_SCALAR,
+
+        /**
+         * The cursor has successfully buffered the entirety of the value on which it is currently positioned, as
+         * requested by an invocation of `fillValue()`.
+         */
+        VALUE_READY,
+
+        /**
+         * The cursor is positioned on a container value.
+         */
+        START_CONTAINER,
+
+        /**
+         * The cursor has reached the end of the current container, and requires an instruction to proceed.
+         */
+        END_CONTAINER
+    }
+
+    /**
+     * Advances the cursor to the next value, skipping the current value (if any).
+     * @return an Event conveying the result of the operation.
+     */
+    Event nextValue();
```

Comment:
Which `Event`s can this return. If I had to guess, I'd say `NEEDS_DATA`, `START_CONTAINER`, `START_SCALAR`, and `END_CONTAINER`. Are those the only ones?

Reply:
I will add documentation that makes it clear which types of events each operation can return.